Source/JavaScriptCore/wasm/WasmCallee.cpp

```diff
@@ -161,6 +164,51 @@ WasmToJSCallee::WasmToJSCallee()
     CalleeRegistry::singleton().registerCallee(this);
 }
 
+IPIntCallee::IPIntCallee(FunctionIPIntMetadataGenerator& generator, size_t index, std::pair<const Name*, RefPtr<NameSection>>&& name)
+    : Callee(Wasm::CompilationMode::IPIntMode, index, WTFMove(name))
+    , m_bytecode(generator.m_bytecode + generator.m_bytecodeOffset)
+    , m_bytecode_len(generator.m_bytecode_len - generator.m_bytecodeOffset)
+    , m_metadata(generator.m_metadata.data())
+    , m_metadata_vec(WTFMove(generator.m_metadata))
+    // size to allocate = 16 + number of stack arguments + number of non-argument locals
+    , m_localSizeToAlloc(roundUpToMultipleOf(16, 16 + generator.m_numArgumentsOnStack + generator.m_numLocals - generator.m_numArguments))
+    , m_numLocals(generator.m_numLocals)
+    , m_numArgumentsOnStack(generator.m_numArgumentsOnStack)
+{
+}
+
+void IPIntCallee::setEntrypoint(CodePtr<WasmEntryPtrTag> entrypoint)
+{
+    ASSERT(!m_entrypoint);
+    m_entrypoint = entrypoint;
+    CalleeRegistry::singleton().registerCallee(this);
+}
+
+RegisterAtOffsetList* IPIntCallee::calleeSaveRegistersImpl()
+{
+    static LazyNeverDestroyed<RegisterAtOffsetList> calleeSaveRegisters;
+    static std::once_flag initializeFlag;
+    std::call_once(initializeFlag, [] {
+        RegisterSet registers;
```

Comment:
Maybe you can define these RegisterSet function in RegisterSet.cpp, like, `RegisterSetBuilder::wasmIPIntCalleeSaveRegisters()`

Reply:
I tried to follow the model that LLInt uses for this function in this case; I can go through and refactor both of them (and maybe DFG, etc.) to set up a function if that would clean up things.